lib/internal/Magento/Framework/DB/Logger/LoggerProxy.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\DB\Logger;
+
+class LoggerProxy extends LoggerAbstract
+{
+    /**
+     * Logger alias param name
+     */
+    const PARAM_ALIAS = 'db_logger_mode';
+
+    /**
+     * Logger log all param name
+     */
+    const PARAM_LOG_ALL = 'db_logger_all';
+
+    /**
+     * Logger query time param name
+     */
+    const PARAM_QUERY_TIME = 'db_logger_query_time';
+
+    /**
+     * Logger call stack param name
+     */
+    const PARAM_CALL_STACK = 'db_logger_stack';
+
+    /**
+     * File logger alias
+     */
+    const LOGGER_ALIAS_FILE = 'file';
+
+    /**
+     * Quiet logger alias
+     */
+    const LOGGER_ALIAS_DISABLED = 'disabled';
+
+    /**
+     * @var LoggerAbstract
+     */
+    private $logger;
+
+    /**
+     * LoggerProxy constructor.
+     * @param File $file
+     * @param Quiet $quiet
+     * @param bool $loggerAlias
+     * @param bool $logAllQueries
+     * @param float $logQueryTime
+     * @param bool $logCallStack
+     */
+    public function __construct(
+        File $file,
+        Quiet $quiet,
+        $loggerAlias,
+        $logAllQueries = true,
+        $logQueryTime = 0.001,
+        $logCallStack = true
+    ) {
+        switch ($loggerAlias) {
+            case self::LOGGER_ALIAS_FILE:
+                $this->logger = $file;
+                break;
+            default:
+                $this->logger = $quiet;
+                break;
+        }
+
+        parent::__construct($logAllQueries, $logQueryTime, $logCallStack);
```

Comment:
I noticed that the ``$logCallStack`` option does not work. I think the reason behind it is that File and Quiet should be constructed with this parameter passed to the constructor. 

This call to the parent method is not really needed because that's File or Quiet should be constructed with these parameters